src/main/java/org/spongepowered/asm/mixin/transformer/ext/Extensions.java

```diff
@@ -176,10 +178,30 @@ public void postApply(ITargetClassContext context) {
      * @param force True to export even if the current environment settings
      *      would normally disable it
      * @param classNode Class to export
+     * @see #export(MixinEnvironment, String, boolean, Supplier) The supplier
+     *      version, which may allow avoiding creating ClassNodes unnecessarily
      */
     public void export(MixinEnvironment env, String name, boolean force, ClassNode classNode) {
+        this.export(env, name, force, Suppliers.ofInstance(classNode));
+    }
+
+    /**
+     * Export class bytecode to disk
+     * 
+     * @param env Environment
+     * @param name Class name
+     * @param force True to export even if the current environment settings
+     *      would normally disable it
+     * @param classNode Supplier providing the class to export
+     */
+    @SuppressWarnings("deprecation")
+    public void export(MixinEnvironment env, String name, boolean force, Supplier<ClassNode> classNode) {
```

Comment:
Why does this need to be made lazy? Only classes that mixin transformed are exported in the first place.

Reply:
Because Mixin also exports bypassed classes:
https://github.com/FabricMC/Mixin/blob/26c8d9c12b1ffa8d1f7cc1e42acb5b113f192b30/src/main/java/org/spongepowered/asm/mixin/transformer/MixinProcessor.java#L301-L312

Which by default will be all accessor mixins:
https://github.com/FabricMC/Mixin/blob/26c8d9c12b1ffa8d1f7cc1e42acb5b113f192b30/src/main/java/org/spongepowered/asm/mixin/transformer/MixinCoprocessorPassthrough.java#L63-L65

There was a reason I didn't deprecate the original method though, it is more likely the `ClassNode` has already been resolved when it's called than not.

Reply:
I'd be ok with changing that behaviour tbh, there's no value in exporting something unchanged.

Reply:
Or if we really care about it for accessor mixins we could just dump the class node their mixininfo already has. Much better than changing the extension API, imo

Reply:
Given we're supporting the odd rogue `Coprocessor`, we can't necessarily say another mod isn't relying on accessors going past their rogue `Extension`. I've left this as is now the custom interface allows `Extensions` to double check whether the `ClassNode` has been loaded (and therefore presumably modified).